dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/dataapproval/DataApprovalLevelDeletionHandler.java

```diff
@@ -27,51 +27,47 @@
  */
 package org.hisp.dhis.dataapproval;
 
+import static org.hisp.dhis.system.deletion.DeletionVeto.ACCEPT;
+
+import lombok.AllArgsConstructor;
+
 import org.hisp.dhis.category.CategoryOptionGroupSet;
 import org.hisp.dhis.system.deletion.DeletionHandler;
+import org.hisp.dhis.system.deletion.DeletionVeto;
 import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Component;
 
 /**
  * @author Jim Grace
  */
+@Component
+@AllArgsConstructor
 public class DataApprovalLevelDeletionHandler
```

Comment:
I guess this handler never was "active" as it wasn't annotated `@Component`.
@jimgrace can you check we still want its checks to happen?

Reply:
@jbee Interesting. I believe the code was correct and active when it was written. The deletion handler `@Component` annotations appear to have been added later [here](https://github.com/dhis2/dhis2-core/commit/0460082f65826af04d1298d97cc7bca2ef3f47af), and it looks like this deletion handler was missed. I don't know why.

Yes, it should be active. Thanks, good catch!

Reply:
Thanks @jimgrace for digging.

Reply:
@jimgrace Now with the handler active this did fail a test and I found that the SQL in the handler referred to a non-existing table. Best I could infer is that we try to prevent deletion of workflows if they are still mapped to levels. I updated the SQL accordingly. Can you check and give feedback if that makes sense to you from the business perspective? Thanks.

Reply:
Ha. It looks like this code is pre-Github, back from when we were using Bazzar. My best guess is that I originally named the link table dataapprovalworkflowmembers based on some other table naming conventions at the time. Then somebody subsequently renamed the table to dataapprovalworkflowlevels (admittedly a better table name), but they didn't change the deletion handler.

Thanks, I agree with your fix.